lib/acts_as_tree.rb

```diff
@@ -141,7 +141,7 @@ def leaves
         class_eval <<-EOV
           def self.leaves
             internal_ids = select(:#{configuration[:foreign_key]}).where(arel_table[:#{configuration[:foreign_key]}].not_eq(nil))
-            where("#{configuration[:primary_key]} NOT IN (\#{internal_ids.to_sql})").default_tree_order
+            where.not(:#{configuration[:primary_key]} => internal_ids).default_tree_order
```

Comment:
This is not backwards compatible, `where.not` is a very recent addition to activerecord.